subsys/net/lib/lwm2m/lwm2m_object.h

```diff
@@ -484,10 +487,16 @@ struct lwm2m_message {
 	struct lwm2m_obj_path path;
 
 	/** CoAP packet data related to the outgoing message */
-	struct coap_packet cpkt;
+	struct coap_packet cpkt[CONFIG_LWM2M_COAP_MAX_BLOCK_NUM];
 
 	/** Buffer data related outgoing message */
-	uint8_t msg_data[MAX_PACKET_SIZE];
+	uint8_t msg_data[CONFIG_LWM2M_COAP_MAX_BLOCK_NUM][MAX_PACKET_SIZE];
```

Comment:
So the idea here is to store the full message within `struct lwm2m_message`, basically multiplying the struct size by tje factor of  `LWM2M_COAP_MAX_BLOCK_NUM`? I don't think this is acceptable, seems like a heavy RAM waste just to cover rare cases when the blockwise transfer is needed.